spec/models/reports/facility_state_spec.rb

```diff
@@ -67,4 +59,103 @@
       end
     end
   end
+
+  context "assigned patients by care states" do
+    describe "under_care, lost_to_follow_up, dead" do
+      it "computes the number of assigned patients correctly with each care state" do
+        facility = create(:facility)
+        _dead_patient = create(:patient, recorded_at: june_2021[:long_ago], assigned_facility: facility, status: "dead")
+        _under_care_patient = create(:patient, recorded_at: june_2021[:under_12_months_ago], assigned_facility: facility)
+        ltfu_patient = create(:patient, recorded_at: june_2021[:over_12_months_ago], assigned_facility: facility)
+        create(:blood_pressure, patient: ltfu_patient, recorded_at: june_2021[:over_12_months_ago])
+        _different_facility_patient = create(:patient, recorded_at: june_2021[:long_ago])
+
+        RefreshMaterializedViews.new.refresh_v2
+        with_reporting_time_zone do
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).under_care).to eq 1
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).lost_to_follow_up).to eq 1
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).dead).to eq 1
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).cumulative_assigned_patients).to eq 3
```

Comment:
@harimohanraj89 had flagged this earlier, should this count exclude dead patients @rsanheim?

Reply:
Reposting my earlier concern: In our dashboard and terminology, "assigned" means "under care" + "LTFU". In some places of our reporting pipeline "assigned" means "under care" + "LTFU" + "dead". Let's nip this confusion in the bud.

Reply:
PS we can remove this assertion in favor of the `cumulative_assigned_patients` specs below, if you'd like.

Reply:
Yep we should exclude `dead` - they are always excluded from any part of the reports where we should cumulative assigned:

<img width="959" alt="image" src="https://user-images.githubusercontent.com/69/127211217-b551b0a5-ae83-4c13-9ac0-519e63b1fef1.png">

Reply:
Thanks folks! Fixed this with 9084c509b.

Reply:
Do we need the total number (under care + LTFU + dead) for anything? If so, are we ok with needing to add assigned + dead to get it?

Reply:
AFAIK, we don't include dead in total assigned. Which, prompts to me think about total registrations actually. Does that need to exclude dead patients as well?

Reply:
That is correct, we should **not** exclude dead patients from registration counts.